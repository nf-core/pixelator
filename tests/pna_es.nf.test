nextflow_pipeline {

    name "Test PNA pipeline with experiment summary"
    script "../main.nf"

    tag "pixelator"
    tag "pixelator/pna"

    test("small") {

        when {
            params {

                pipelines_testdata_base_path           = "https://raw.githubusercontent.com/nf-core/test-datasets/pixelator/"
                input                                  = "$pipelines_testdata_base_path/samplesheet/pna/samplesheet_pna_es.csv"
                input_basedir                          = "$pipelines_testdata_base_path/testdata/pna/"
                outdir                                 = "$outputDir"

                pna_graph_component_size_min_threshold = 100
                skip_experiment_summary                = false
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = []

            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_pixelator_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }

    test("small - stub") {

        options "-stub"
        tag "stub"

        when {
            params {
                pipelines_testdata_base_path = "$projectDir/../nf-core-pixelator-datasets/"
                input                        = "$pipelines_testdata_base_path/samplesheet/pna/samplesheet_pna_es.csv"
                input_basedir                = "$pipelines_testdata_base_path/testdata/pna/"
                outdir                       = "$outputDir"
                skip_experiment_summary      = false
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = []
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
}
