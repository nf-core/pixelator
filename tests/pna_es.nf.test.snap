{
    "small": {
        "content": [
            22,
            {
                "EXPERIMENT_SUMMARY": {
                    "experiment-summary": "0.4.3"
                },
                "PIXELATOR_PNA_AMPLICON": {
                    "pixelator": "0.21.2"
                },
                "PIXELATOR_PNA_ANALYSIS": {
                    "pixelator": "0.21.2"
                },
                "PIXELATOR_PNA_COLLAPSE": {
                    "pixelator": "0.21.2"
                },
                "PIXELATOR_PNA_COMBINE_COLLAPSE": {
                    "pixelator": "0.21.2"
                },
                "PIXELATOR_PNA_DEMUX": {
                    "pixelator": "0.21.2"
                },
                "PIXELATOR_PNA_DENOISE": {
                    "pixelator": "0.21.2"
                },
                "PIXELATOR_PNA_GRAPH": {
                    "pixelator": "0.21.2"
                },
                "PIXELATOR_PNA_LAYOUT": {
                    "pixelator": "0.21.2"
                },
                "PIXELATOR_PNA_REPORT": {
                    "pixelator": "0.21.2"
                },
                "Workflow": {
                    "nf-core/pixelator": "v2.0.0"
                }
            },
            [
                "pipeline_info",
                "pipeline_info/nf_core_pixelator_software_mqc_versions.yml",
                "pixelator",
                "pixelator/PNA055_Sample07_filtered_S7.layout.pxl",
                "pixelator/PNA055_Sample07_unfiltered_S7.layout.pxl",
                "pixelator/amplicon",
                "pixelator/amplicon/PNA055_Sample07_filtered_S7.meta.json",
                "pixelator/amplicon/PNA055_Sample07_filtered_S7.report.json",
                "pixelator/amplicon/PNA055_Sample07_unfiltered_S7.meta.json",
                "pixelator/amplicon/PNA055_Sample07_unfiltered_S7.report.json",
                "pixelator/analysis",
                "pixelator/analysis/PNA055_Sample07_filtered_S7.meta.json",
                "pixelator/analysis/PNA055_Sample07_filtered_S7.report.json",
                "pixelator/analysis/PNA055_Sample07_unfiltered_S7.meta.json",
                "pixelator/analysis/PNA055_Sample07_unfiltered_S7.report.json",
                "pixelator/collapse",
                "pixelator/collapse/PNA055_Sample07_filtered_S7.collapse.m1.part_000.meta.json",
                "pixelator/collapse/PNA055_Sample07_filtered_S7.collapse.m1.part_000.report.json",
                "pixelator/collapse/PNA055_Sample07_filtered_S7.collapse.m2.part_000.meta.json",
                "pixelator/collapse/PNA055_Sample07_filtered_S7.collapse.m2.part_000.report.json",
                "pixelator/collapse/PNA055_Sample07_filtered_S7.collapse.parquet",
                "pixelator/collapse/PNA055_Sample07_filtered_S7.meta.json",
                "pixelator/collapse/PNA055_Sample07_filtered_S7.report.json",
                "pixelator/collapse/PNA055_Sample07_unfiltered_S7.collapse.m1.part_000.meta.json",
                "pixelator/collapse/PNA055_Sample07_unfiltered_S7.collapse.m1.part_000.report.json",
                "pixelator/collapse/PNA055_Sample07_unfiltered_S7.collapse.m2.part_000.meta.json",
                "pixelator/collapse/PNA055_Sample07_unfiltered_S7.collapse.m2.part_000.report.json",
                "pixelator/collapse/PNA055_Sample07_unfiltered_S7.collapse.parquet",
                "pixelator/collapse/PNA055_Sample07_unfiltered_S7.meta.json",
                "pixelator/collapse/PNA055_Sample07_unfiltered_S7.report.json",
                "pixelator/demux",
                "pixelator/demux/PNA055_Sample07_filtered_S7.meta.json",
                "pixelator/demux/PNA055_Sample07_filtered_S7.report.json",
                "pixelator/demux/PNA055_Sample07_unfiltered_S7.meta.json",
                "pixelator/demux/PNA055_Sample07_unfiltered_S7.report.json",
                "pixelator/denoise",
                "pixelator/denoise/PNA055_Sample07_filtered_S7.meta.json",
                "pixelator/denoise/PNA055_Sample07_filtered_S7.report.json",
                "pixelator/denoise/PNA055_Sample07_unfiltered_S7.meta.json",
                "pixelator/denoise/PNA055_Sample07_unfiltered_S7.report.json",
                "pixelator/experiment-summary.html",
                "pixelator/graph",
                "pixelator/graph/PNA055_Sample07_filtered_S7.meta.json",
                "pixelator/graph/PNA055_Sample07_filtered_S7.report.json",
                "pixelator/graph/PNA055_Sample07_unfiltered_S7.meta.json",
                "pixelator/graph/PNA055_Sample07_unfiltered_S7.report.json",
                "pixelator/layout",
                "pixelator/layout/PNA055_Sample07_filtered_S7.meta.json",
                "pixelator/layout/PNA055_Sample07_filtered_S7.report.json",
                "pixelator/layout/PNA055_Sample07_unfiltered_S7.meta.json",
                "pixelator/layout/PNA055_Sample07_unfiltered_S7.report.json",
                "pixelator/logs",
                "pixelator/logs/PNA055_Sample07_filtered_S7",
                "pixelator/logs/PNA055_Sample07_filtered_S7.pixelator-combine-collapse.log",
                "pixelator/logs/PNA055_Sample07_filtered_S7.pixelator-report.log",
                "pixelator/logs/PNA055_Sample07_filtered_S7/PNA055_Sample07_filtered_S7.pixelator-amplicon.log",
                "pixelator/logs/PNA055_Sample07_filtered_S7/PNA055_Sample07_filtered_S7.pixelator-analysis.log",
                "pixelator/logs/PNA055_Sample07_filtered_S7/PNA055_Sample07_filtered_S7.pixelator-collapse.log",
                "pixelator/logs/PNA055_Sample07_filtered_S7/PNA055_Sample07_filtered_S7.pixelator-demux.log",
                "pixelator/logs/PNA055_Sample07_filtered_S7/PNA055_Sample07_filtered_S7.pixelator-denoise.log",
                "pixelator/logs/PNA055_Sample07_filtered_S7/PNA055_Sample07_filtered_S7.pixelator-graph.log",
                "pixelator/logs/PNA055_Sample07_filtered_S7/PNA055_Sample07_filtered_S7.pixelator-layout.log",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7.pixelator-combine-collapse.log",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7.pixelator-report.log",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7/PNA055_Sample07_unfiltered_S7.pixelator-amplicon.log",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7/PNA055_Sample07_unfiltered_S7.pixelator-analysis.log",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7/PNA055_Sample07_unfiltered_S7.pixelator-collapse.log",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7/PNA055_Sample07_unfiltered_S7.pixelator-demux.log",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7/PNA055_Sample07_unfiltered_S7.pixelator-denoise.log",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7/PNA055_Sample07_unfiltered_S7.pixelator-graph.log",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7/PNA055_Sample07_unfiltered_S7.pixelator-layout.log",
                "pixelator/report",
                "pixelator/report/PNA055_Sample07_filtered_S7.qc-report.html",
                "pixelator/report/PNA055_Sample07_unfiltered_S7.qc-report.html"
            ],
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-27T08:04:38.973408779"
    },
    "small - stub": {
        "content": [
            22,
            [
                "pipeline_info",
                "pipeline_info/nf_core_pixelator_software_mqc_versions.yml",
                "pixelator",
                "pixelator/PNA055_Sample07_filtered_S7.pxl",
                "pixelator/PNA055_Sample07_unfiltered_S7.pxl",
                "pixelator/amplicon",
                "pixelator/amplicon/PNA055_Sample07_filtered_S7.meta.json",
                "pixelator/amplicon/PNA055_Sample07_filtered_S7.report.json",
                "pixelator/amplicon/PNA055_Sample07_unfiltered_S7.meta.json",
                "pixelator/amplicon/PNA055_Sample07_unfiltered_S7.report.json",
                "pixelator/analysis",
                "pixelator/analysis/PNA055_Sample07_filtered_S7.meta.json",
                "pixelator/analysis/PNA055_Sample07_filtered_S7.report.json",
                "pixelator/analysis/PNA055_Sample07_unfiltered_S7.meta.json",
                "pixelator/analysis/PNA055_Sample07_unfiltered_S7.report.json",
                "pixelator/collapse",
                "pixelator/collapse/PNA055_Sample07_filtered_S7.collapse.meta.json",
                "pixelator/collapse/PNA055_Sample07_filtered_S7.collapse.parquet",
                "pixelator/collapse/PNA055_Sample07_filtered_S7.demux.m1.part_000.meta.json",
                "pixelator/collapse/PNA055_Sample07_filtered_S7.demux.m1.part_000.report.json",
                "pixelator/collapse/PNA055_Sample07_filtered_S7.demux.m2.part_000.meta.json",
                "pixelator/collapse/PNA055_Sample07_filtered_S7.demux.m2.part_000.report.json",
                "pixelator/collapse/PNA055_Sample07_filtered_S7.report.json",
                "pixelator/collapse/PNA055_Sample07_unfiltered_S7.collapse.meta.json",
                "pixelator/collapse/PNA055_Sample07_unfiltered_S7.collapse.parquet",
                "pixelator/collapse/PNA055_Sample07_unfiltered_S7.demux.m1.part_000.meta.json",
                "pixelator/collapse/PNA055_Sample07_unfiltered_S7.demux.m1.part_000.report.json",
                "pixelator/collapse/PNA055_Sample07_unfiltered_S7.demux.m2.part_000.meta.json",
                "pixelator/collapse/PNA055_Sample07_unfiltered_S7.demux.m2.part_000.report.json",
                "pixelator/collapse/PNA055_Sample07_unfiltered_S7.report.json",
                "pixelator/demux",
                "pixelator/demux/PNA055_Sample07_filtered_S7.meta.json",
                "pixelator/demux/PNA055_Sample07_filtered_S7.report.json",
                "pixelator/demux/PNA055_Sample07_unfiltered_S7.meta.json",
                "pixelator/demux/PNA055_Sample07_unfiltered_S7.report.json",
                "pixelator/denoise",
                "pixelator/denoise/PNA055_Sample07_filtered_S7.meta.json",
                "pixelator/denoise/PNA055_Sample07_filtered_S7.report.json",
                "pixelator/denoise/PNA055_Sample07_unfiltered_S7.meta.json",
                "pixelator/denoise/PNA055_Sample07_unfiltered_S7.report.json",
                "pixelator/experiment-summary.html",
                "pixelator/graph",
                "pixelator/graph/PNA055_Sample07_filtered_S7.meta.json",
                "pixelator/graph/PNA055_Sample07_filtered_S7.report.json",
                "pixelator/graph/PNA055_Sample07_unfiltered_S7.meta.json",
                "pixelator/graph/PNA055_Sample07_unfiltered_S7.report.json",
                "pixelator/layout",
                "pixelator/layout/PNA055_Sample07_filtered_S7.meta.json",
                "pixelator/layout/PNA055_Sample07_filtered_S7.report.json",
                "pixelator/layout/PNA055_Sample07_unfiltered_S7.meta.json",
                "pixelator/layout/PNA055_Sample07_unfiltered_S7.report.json",
                "pixelator/logs",
                "pixelator/logs/PNA055_Sample07_filtered_S7",
                "pixelator/logs/PNA055_Sample07_filtered_S7.pixelator-combine-collapse.log",
                "pixelator/logs/PNA055_Sample07_filtered_S7.pixelator-report.log",
                "pixelator/logs/PNA055_Sample07_filtered_S7/PNA055_Sample07_filtered_S7.demux.m1.part_000.pixelator-collapse.log",
                "pixelator/logs/PNA055_Sample07_filtered_S7/PNA055_Sample07_filtered_S7.demux.m2.part_000.pixelator-collapse.log",
                "pixelator/logs/PNA055_Sample07_filtered_S7/PNA055_Sample07_filtered_S7.pixelator-amplicon.log",
                "pixelator/logs/PNA055_Sample07_filtered_S7/PNA055_Sample07_filtered_S7.pixelator-analysis.log",
                "pixelator/logs/PNA055_Sample07_filtered_S7/PNA055_Sample07_filtered_S7.pixelator-demux.log",
                "pixelator/logs/PNA055_Sample07_filtered_S7/PNA055_Sample07_filtered_S7.pixelator-denoise.log",
                "pixelator/logs/PNA055_Sample07_filtered_S7/PNA055_Sample07_filtered_S7.pixelator-graph.log",
                "pixelator/logs/PNA055_Sample07_filtered_S7/PNA055_Sample07_filtered_S7.pixelator-layout.log",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7.pixelator-combine-collapse.log",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7.pixelator-report.log",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7/PNA055_Sample07_unfiltered_S7.demux.m1.part_000.pixelator-collapse.log",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7/PNA055_Sample07_unfiltered_S7.demux.m2.part_000.pixelator-collapse.log",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7/PNA055_Sample07_unfiltered_S7.pixelator-amplicon.log",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7/PNA055_Sample07_unfiltered_S7.pixelator-analysis.log",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7/PNA055_Sample07_unfiltered_S7.pixelator-demux.log",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7/PNA055_Sample07_unfiltered_S7.pixelator-denoise.log",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7/PNA055_Sample07_unfiltered_S7.pixelator-graph.log",
                "pixelator/logs/PNA055_Sample07_unfiltered_S7/PNA055_Sample07_unfiltered_S7.pixelator-layout.log",
                "pixelator/report",
                "pixelator/report/PNA055_Sample07_filtered_S7.report.html",
                "pixelator/report/PNA055_Sample07_unfiltered_S7.report.html"
            ],
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-25T11:37:33.484947103"
    }
}